CRC errors

I'm having problems with dimmer changes (turned on by switch) not updating the GUI. Checked the log and gets a lot of CRC errors, this works on normal switches (Fibaro wall plug). The dimmer is a "FGD212 Dimmer 2" but I have some Qubinos that doesn't work either.

> 21:06:08.977 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains E0 BC  but should be E0 00
21:06:09.702 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains 77 BE  but should be 77 00
21:06:14.700 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains 17 C1  but should be 17 00

what could it be? Feels like some kind of bug to me.

Please provide a drug log - it's impossible to debug without this. 

Sent from my iPhone

> On 17 Jan 2017, at 02:40, Edgesystems <notifications@github.com> wrote:
> 
> I'm having problems with dimmer changes (turned on by switch) not updating the GUI. Checked the log and gets a lot of CRC errors, this works on normal switches (Fibaro wall plug). The dimmer is a "FGD212 Dimmer 2" but I have some Qubinos that doesn't work either.
> 
> 21:06:08.977 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains E0 BC but should be E0 00
> 21:06:09.702 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains 77 BE but should be 77 00
> 21:06:14.700 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains 17 C1 but should be 17 00
> 
> what could it be? Feels like some kind of bug to me.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

This is the output when I turn the dimmer on via the switch.

```
2017-01-18 18:56:46.865 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 0E 00 04 00 04 07 56 01 26 03 19 BD E0 CF 0F
2017-01-18 18:56:46.867 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:46.869 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 0E 00 04 00 04 07 56 01 26 03 19 BD E0 CF 0F
2017-01-18 18:56:46.870 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 0E 00 04 00 04 07 56 01 26 03 19 BD E0 CF 0F
2017-01-18 18:56:46.871 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 07 56 01 26 03 19 BD E0 CF
2017-01-18 18:56:46.872 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Application Command Request (ALIVE:DYNAMIC_VALUES)
2017-01-18 18:56:46.873 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Incoming command class CRC_16_ENCAP
2017-01-18 18:56:46.873 [DEBUG] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: Received CRC 16 Encapsulation Request
2017-01-18 18:56:46.874 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains E0 CF  but should be E0 00
2017-01-18 18:56:46.875 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Sent Message: class=SendData[0x13], type=Request[0x00], priority=Get, dest=4, callback=167, payload=04 03 71 04 08
2017-01-18 18:56:46.877 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Recv Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 07 56 01 26 03 19 BD E0 CF
2017-01-18 18:56:46.880 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: class=ApplicationCommandHandler, callback id=167, expected=ApplicationCommandHandler, cancelled=false        transaction complete!
2017-01-18 18:56:46.881 [DEBUG] [ve.internal.protocol.ZWaveController] - Notifying event listeners: ZWaveTransactionCompletedEvent
2017-01-18 18:56:46.883 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - DYNAMIC_VALUES: Transaction complete (SendData:Request) success(true)
2017-01-18 18:56:46.885 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - checking initialisation queue. Queue size 16.
2017-01-18 18:56:46.887 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - message removed from queue. Queue size 15.
2017-01-18 18:56:46.889 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - DYNAMIC_VALUES: queue length(15), free to send(true)
2017-01-18 18:56:46.889 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Initialisation retry timer started 80000
2017-01-18 18:56:46.890 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - queued packet. Queue length is 15
2017-01-18 18:56:46.891 [DEBUG] [ocol.ZWaveController$ZWaveSendThread] - Took message from queue for sending. Queue length = 0
2017-01-18 18:56:46.892 [DEBUG] [ve.internal.protocol.ZWaveController] - Message queued. Queue length = 0. Queue={}
2017-01-18 18:56:46.892 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 0A 00 13 04 03 71 04 09 25 A8 10
2017-01-18 18:56:46.893 [DEBUG] [ing.zwave.handler.ZWaveSerialHandler] - NODE 4: Sending REQUEST Message = 01 0A 00 13 04 03 71 04 09 25 A8 10
2017-01-18 18:56:46.894 [DEBUG] [ding.zwave.handler.ZWaveThingHandler] - NODE 4: Got an event from Z-Wave network: ZWaveTransactionCompletedEvent
2017-01-18 18:56:46.900 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 04 01 13 01 E8
2017-01-18 18:56:46.906 [DEBUG] [ocol.ZWaveController$ZWaveSendThread] - NODE 4: Response processed after 0ms/4677ms.
2017-01-18 18:56:46.907 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:46.909 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 04 01 13 01 E8
2017-01-18 18:56:46.910 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 04 01 13 01 E8
2017-01-18 18:56:46.911 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=SendData[0x13], type=Response[0x01], priority=High, dest=255, callback=0, payload=01
2017-01-18 18:56:46.913 [DEBUG] [l.serialmessage.SendDataMessageClass] - NODE 4: Sent Data successfully placed on stack.
2017-01-18 18:56:47.311 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 11 00 04 00 04 0A 56 01 31 05 04 22 00 08 F8 D3 CF 4D
2017-01-18 18:56:47.314 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:47.317 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 11 00 04 00 04 0A 56 01 31 05 04 22 00 08 F8 D3 CF 4D
2017-01-18 18:56:47.318 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 11 00 04 00 04 0A 56 01 31 05 04 22 00 08 F8 D3 CF 4D
2017-01-18 18:56:47.320 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 0A 56 01 31 05 04 22 00 08 F8 D3 CF
2017-01-18 18:56:47.321 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Application Command Request (ALIVE:DYNAMIC_VALUES)
2017-01-18 18:56:47.322 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Incoming command class CRC_16_ENCAP
2017-01-18 18:56:47.323 [DEBUG] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: Received CRC 16 Encapsulation Request
2017-01-18 18:56:47.323 [ERROR] [.ZWaveCRC16EncapsulationCommandClass] - NODE 4: CRC check failed message contains D3 CF  but should be D3 00
2017-01-18 18:56:47.325 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Sent Message: class=SendData[0x13], type=Request[0x00], priority=Get, dest=4, callback=168, payload=04 03 71 04 09
2017-01-18 18:56:47.326 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Recv Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 0A 56 01 31 05 04 22 00 08 F8 D3 CF
2017-01-18 18:56:47.327 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: class=ApplicationCommandHandler, callback id=168, expected=ApplicationCommandHandler, cancelled=false        transaction complete!
2017-01-18 18:56:47.328 [DEBUG] [ve.internal.protocol.ZWaveController] - Notifying event listeners: ZWaveTransactionCompletedEvent
2017-01-18 18:56:47.328 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - DYNAMIC_VALUES: Transaction complete (SendData:Request) success(true)
2017-01-18 18:56:47.329 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - checking initialisation queue. Queue size 15.
2017-01-18 18:56:47.330 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - message removed from queue. Queue size 14.
2017-01-18 18:56:47.332 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - DYNAMIC_VALUES: queue length(14), free to send(true)
2017-01-18 18:56:47.334 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Initialisation retry timer started 80000
2017-01-18 18:56:47.335 [DEBUG] [alization.ZWaveNodeInitStageAdvancer] - NODE 4: Node advancer - queued packet. Queue length is 14
2017-01-18 18:56:47.338 [DEBUG] [ocol.ZWaveController$ZWaveSendThread] - Took message from queue for sending. Queue length = 0
2017-01-18 18:56:47.338 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 0A 00 13 04 03 71 04 04 25 A9 1C
2017-01-18 18:56:47.339 [DEBUG] [ing.zwave.handler.ZWaveSerialHandler] - NODE 4: Sending REQUEST Message = 01 0A 00 13 04 03 71 04 04 25 A9 1C
2017-01-18 18:56:47.339 [DEBUG] [ve.internal.protocol.ZWaveController] - Message queued. Queue length = 0. Queue={}
2017-01-18 18:56:47.341 [DEBUG] [ding.zwave.handler.ZWaveThingHandler] - NODE 4: Got an event from Z-Wave network: ZWaveTransactionCompletedEvent
2017-01-18 18:56:47.346 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 04 01 13 01 E8
2017-01-18 18:56:47.356 [DEBUG] [ocol.ZWaveController$ZWaveSendThread] - NODE 4: Response processed after 0ms/4677ms.
2017-01-18 18:56:47.357 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:47.359 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 04 01 13 01 E8
2017-01-18 18:56:47.360 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 04 01 13 01 E8
2017-01-18 18:56:47.361 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=SendData[0x13], type=Response[0x01], priority=High, dest=255, callback=0, payload=01
2017-01-18 18:56:47.363 [DEBUG] [l.serialmessage.SendDataMessageClass] - NODE 4: Sent Data successfully placed on stack.
2017-01-18 18:56:47.958 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 18 00 13 A9 00 00 69 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 03 00 00 F6
2017-01-18 18:56:47.960 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:47.964 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 1A 00 13 A9 00 00 69 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 03 00 00 00 00 F4
2017-01-18 18:56:47.967 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 1A 00 13 A9 00 00 69 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 03 00 00 00 00 F4
2017-01-18 18:56:47.971 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=A9 00 00 69 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 03 00 00
2017-01-18 18:56:47.973 [DEBUG] [l.serialmessage.SendDataMessageClass] - NODE 4: SendData Request. CallBack ID = 169, Status = Transmission complete and ACK received(0)
2017-01-18 18:56:47.977 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Sent Message: class=SendData[0x13], type=Request[0x00], priority=Get, dest=4, callback=169, payload=04 03 71 04 04
2017-01-18 18:56:47.981 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Recv Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=A9 00 00 69 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 03 00 00
2017-01-18 18:56:47.983 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: class=SendData, callback id=169, expected=ApplicationCommandHandler, cancelled=false      MISMATCH
2017-01-18 18:56:48.869 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 18 00 13 A9 00 00 98 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 01 00 00 05
2017-01-18 18:56:48.936 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:48.938 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 1A 00 13 A9 00 00 98 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 01 00 00 00 00 07
2017-01-18 18:56:48.941 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 1A 00 13 A9 00 00 98 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 01 00 00 00 00 07
2017-01-18 18:56:48.945 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=A9 00 00 98 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 01 00 00
2017-01-18 18:56:48.948 [DEBUG] [l.serialmessage.SendDataMessageClass] - NODE 4: SendData Request. CallBack ID = 169, Status = Transmission complete and ACK received(0)
2017-01-18 18:56:48.951 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Sent Message: class=SendData[0x13], type=Request[0x00], priority=Get, dest=4, callback=169, payload=04 03 71 04 04
2017-01-18 18:56:48.954 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: Recv Message: class=SendData[0x13], type=Request[0x00], priority=High, dest=255, callback=0, payload=A9 00 00 98 01 CF 7F 7F 7F 7F 01 01 03 0D 00 00 00 01 01 00 00
2017-01-18 18:56:48.957 [DEBUG] [.serialmessage.ZWaveCommandProcessor] - Checking transaction complete: class=SendData, callback id=169, expected=ApplicationCommandHandler, cancelled=false      MISMATCH
2017-01-18 18:56:48.987 [DEBUG] [WaveSerialHandler$ZWaveReceiveThread] - Receive Message = 01 0B 00 04 00 04 04 71 05 00 00 CF 4B
2017-01-18 18:56:48.988 [DEBUG] [ve.internal.protocol.ZWaveController] - Receive queue TAKE: Length=0
2017-01-18 18:56:48.990 [DEBUG] [wave.internal.protocol.SerialMessage] - Assembled message buffer = 01 0B 00 04 00 04 04 71 05 00 00 CF 4B
2017-01-18 18:56:48.991 [DEBUG] [ve.internal.protocol.ZWaveController] - Process Message = 01 0B 00 04 00 04 04 71 05 00 00 CF 4B
2017-01-18 18:56:48.993 [DEBUG] [ve.internal.protocol.ZWaveController] - Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 04 71 05 00 00 CF
2017-01-18 18:56:48.994 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Application Command Request (ALIVE:DYNAMIC_VALUES)
2017-01-18 18:56:48.995 [DEBUG] [ssage.ApplicationCommandMessageClass] - NODE 4: Incoming command class ALARM
2017-01-18 18:56:48.997 [DEBUG] [.commandclass.ZWaveAlarmCommandClass] - NODE 4: Received ALARM command V5
2017-01-18 18:56:48.998 [DEBUG] [.commandclass.ZWaveAlarmCommandClass] - NODE 4: Process NOTIFICATION_REPORT V5
2017-01-18 18:56:49.000 [ERROR] [ssage.ApplicationCommandMessageClass] - Error processing frame: Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 04 71 05 00 00 CF  >> Attempt to read message payload out of bounds: Message: class=ApplicationCommandHandler[0x04], type=Request[0x00], priority=High, dest=255, callback=0, payload=00 04 04 71 05 00 00 CF  (8)

```

--- a/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveCRC16EncapsulationCommandClass.java
+++ b/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveCRC16EncapsulationCommandClass.java
@@ -12,10 +12,10 @@ import java.nio.ByteBuffer;
 import java.util.Arrays;
 
 import org.openhab.binding.zwave.internal.protocol.SerialMessage;
-import org.openhab.binding.zwave.internal.protocol.ZWaveSerialMessageException;
 import org.openhab.binding.zwave.internal.protocol.ZWaveController;
 import org.openhab.binding.zwave.internal.protocol.ZWaveEndpoint;
 import org.openhab.binding.zwave.internal.protocol.ZWaveNode;
+import org.openhab.binding.zwave.internal.protocol.ZWaveSerialMessageException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -60,7 +60,7 @@ public class ZWaveCRC16EncapsulationCommandClass extends ZWaveCommandClass {
 
     /**
      * {@inheritDoc}
-     * 
+     *
      * @throws ZWaveSerialMessageException
      */
     @Override
@@ -90,9 +90,9 @@ public class ZWaveCRC16EncapsulationCommandClass extends ZWaveCommandClass {
 
         // calculate CRC
         byte[] payload = serialMessage.getMessagePayload();
-        byte[] messageCrc = Arrays.copyOfRange(payload, payload.length - 2, payload.length);
-        byte[] tocheck = Arrays.copyOfRange(payload, offset - 2, payload.length - 2);
-
+        int length = payload[2];
+        byte[] messageCrc = Arrays.copyOfRange(payload, length + 1, length + 3);
+        byte[] tocheck = Arrays.copyOfRange(payload, offset - 2, offset + length - 4);
         short calculatedCrc = crc_ccit(tocheck);
         // check if messageCrc = calculatedCrc
         ByteBuffer byteBuffer = ByteBuffer.allocate(2);

--- a/src/test/java/org/openhab/binding/zwave/test/internal/protocol/commandclass/ZWaveCrc16EncapsulationCommandClassTest.java
+++ b/src/test/java/org/openhab/binding/zwave/test/internal/protocol/commandclass/ZWaveCrc16EncapsulationCommandClassTest.java
@@ -2,8 +2,10 @@ package org.openhab.binding.zwave.test.internal.protocol.commandclass;
 
 import static org.junit.Assert.assertEquals;
 
+import java.math.BigDecimal;
 import java.util.List;
 
+import org.junit.Ignore;
 import org.junit.Test;
 import org.openhab.binding.zwave.internal.protocol.commandclass.ZWaveCommandClass.CommandClass;
 import org.openhab.binding.zwave.internal.protocol.event.ZWaveCommandClassValueEvent;
@@ -16,8 +18,18 @@ import org.openhab.binding.zwave.internal.protocol.event.ZWaveEvent;
  *
  */
 public class ZWaveCrc16EncapsulationCommandClassTest extends ZWaveCommandClassTest {
+    // This test uses the data from the spec
+    @Ignore
     @Test
-    public void ProcessMessage() {
+    public void ProcessMessageZWaveSpec() {
+        byte[] packetData = { 0x01, 0x0d, 0x00, 0x04, 0x00, 0x07, 0x07, 0x56, 0x01, 0x20, 0x02, 0x4d, 0x26,
+                (byte) 0xe8 };
+
+        processCommandClassMessage(packetData);
+    }
+
+    @Test
+    public void ProcessMessage1() {
         byte[] packetData = { 0x01, 0x0d, 0x00, 0x04, 0x00, 0x07, 0x07, 0x56, 0x01, 0x20, 0x03, 0x00, (byte) 0x8c, 0x58,
                 0x56 };
 
@@ -27,5 +39,18 @@ public class ZWaveCrc16EncapsulationCommandClassTest extends ZWaveCommandClassTe
         assertEquals(event.getCommandClass(), CommandClass.BASIC);
         assertEquals(event.getEndpoint(), 0);
         assertEquals(event.getValue(), 0);
+    }
+
+    @Test
+    public void ProcessMessage2() {
+        byte[] packetData = { 0x01, 0x11, 0x00, 0x04, 0x00, 0x04, 0x0A, 0x56, 0x01, 0x31, 0x05, 0x04, 0x22, 0x00, 0x08,
+                (byte) 0xF8, (byte) 0xD3, (byte) 0xCF, 0x4D };
+
+        List<ZWaveEvent> events = processCommandClassMessage(packetData);
+        assertEquals(events.size(), 1);
+        ZWaveCommandClassValueEvent event = (ZWaveCommandClassValueEvent) events.get(0);
+        assertEquals(event.getCommandClass(), CommandClass.SENSOR_MULTILEVEL);
+        assertEquals(event.getEndpoint(), 0);
+        assertEquals(event.getValue(), new BigDecimal("0.8"));
     }
 }

